### PR TITLE
Add extra-args configs to kubernetes-worker charm

### PR DESCRIPTION
**Pull request by Cynerva**
**What this PR does / why we need it**:

This adds `kubelet-extra-args` and `proxy-extra-args` config options to the kubernetes-worker charm, which allow operators to pass arbitrary configuraton to the kubelet and kube-proxy services.

This also fixes a bug where the equivalent config options on kubernetes-master (`api-extra-args`, `controller-manager-extra-args`, `scheduler-extra-args`) don't remove arguments from the services when they are removed from the config.

**Release note**:

```release-note
Add extra-args configs to kubernetes-worker charm
```

### CODE REVIEW

File: cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py [hunk 1]
```diff
@@ -469,44 +468,92 @@ def create_config(server, creds):
...
+    privileged = is_state('kubernetes-worker.privileged')
+    kubelet_opts['allow-privileged'] = 'true' if privileged else 'false'
+
+    if is_state('kubernetes-worker.gpu.enabled'):
+        if get_version('kubelet') < (1, 6):
+            hookenv.log('Adding --experimental-nvidia-gpus=1 to kubelet')
+            kubelet_opts['experimental-nvidia-gpus'] = '1'
+        else:
+            hookenv.log('Adding --feature-gates=Accelerators=true to kubelet')
+            kubelet_opts['feature-gates'] = 'Accelerators=true'
```
Comment (Cynerva) [inline cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py]: I need to test that the GPU functionality still works with this change. I don't think our automated tests cover it.

Comment (Cynerva) [reply cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py]: Tested with a GPU worker. This worked how I would expect, so we should be good.

**Comment by Cynerva** [comment]
/assign @ktsakalozos 

Can you review this please? This should fix the bug that our CI tests are failing on. Thanks!

**Comment by ktsakalozos** [comment]
Less code, less room for bugs. +1

**Pull request merged**